backend/schemas/user.py

```
@@ -40,21 +43,141 @@ class UserCreate(UserBase):
+            
+        # Strip whitespace first
+        v = v.strip()
+        
+        # Check for empty string after stripping
+        if not v:
+            return None
+            
+        # 1. HTML tag detection using regex (more robust than string matching)
+        html_pattern = re.compile(r'<[^>]*>', re.IGNORECASE)
+        if html_pattern.search(v):
+            raise ValueError("HTML tags are not allowed in names.")
```

Ruff (0.13.2)

Avoid specifying long messages outside the exception class

(TRY003)